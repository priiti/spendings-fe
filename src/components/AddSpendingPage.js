import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addSpending } from './../actions/spendings';
import SpendingsForm from './SpendingsForm';

const AddSpendingPage = (props) => (
  <div>
    <h1>Add new spending</h1>
    <SpendingsForm
      onSubmit={(spending) => {
        props.addSpending(spending);
        props.history.push('/');
      }} 
    />
  </div>
);

const mapStateToProps = (state) => {
  return {};
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    addSpending
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AddSpendingPage);
