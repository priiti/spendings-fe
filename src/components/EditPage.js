import React from 'react';

const EditPage = (props) => {
  console.log(props);

  return (
    <div>
      Editing expense with id {props.match.params.id} page
    </div>
  );
};

export default EditPage;
