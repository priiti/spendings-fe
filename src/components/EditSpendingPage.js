import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SpendingsForm from './../components/SpendingsForm';
import { editSpending, removeSpending } from './../actions/spendings';

const EditSpendingPage = (props) => {
  const { id } = props.match.params;
  
  return (
    <div>
      <SpendingsForm
        spending={props.spending}
        onSubmit={(spending) => {
          props.editSpending(id, spending);
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.removeSpending({ id });
        props.history.push('/');
      }}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    spending: state.spendings.find((spending) => spending.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    editSpending,
    removeSpending
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(EditSpendingPage);
