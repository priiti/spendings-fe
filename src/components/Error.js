import React from 'react';

const Error = (props) => {
  const { error } = props;

  return (
    <div>
      <p>{error}</p>
    </div>
  );
};

export default Error;
