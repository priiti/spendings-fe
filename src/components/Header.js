import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>ğŸ’° Spendings manager</h1>
    <NavLink to="/" activeClassName="is-active" exact>Homepage</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
);

export default Header;
