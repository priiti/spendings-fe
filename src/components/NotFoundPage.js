import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    404 - Not found - <Link to="/">Dashboard</Link>
  </div>
);

export default NotFoundPage;
