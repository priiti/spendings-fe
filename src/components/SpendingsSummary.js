import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getFilteredSpendings from './../selectors/spendings';
import selectSpendingsTotal from './../selectors/spendings-total';

export const SpendingsSummary = (props) => {
  const {
    spendingsCount,
    spendingsTotal
  } = props;

  const spendingsWord = spendingsCount === 1 ? 'spending' : 'spendings';
  const formattedSpendingsTotalSum = numeral(spendingsTotal / 100).format('$0.00');

  return (
    <div>
      {spendingsCount > 0 &&
        <h1>Viewing {spendingsCount} {spendingsWord} totaling {formattedSpendingsTotalSum}</h1>
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleSpendings = getFilteredSpendings(state.spendings, state.filters);
  return {
    spendingsCount: visibleSpendings.length,
    spendingsTotal: selectSpendingsTotal(visibleSpendings),
  };
};

export default connect(mapStateToProps, {})(SpendingsSummary);
