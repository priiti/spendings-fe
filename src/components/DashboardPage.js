import React from 'react';
import SpendingsList from './../components/SpendingsList';
import SpendingsListFiltersContainer from './../components/SpendingsListFiltersContainer';
import SpendingsSummary from './SpendingsSummary';

const DashboardPage = () => (
  <div>
    <SpendingsSummary />
    <SpendingsListFiltersContainer />
    <SpendingsList />
  </div>
);

export default DashboardPage;
