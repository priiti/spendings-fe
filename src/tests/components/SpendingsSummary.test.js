import React from 'react';
import { shallow } from 'enzyme';
import { SpendingsSummary } from './../../components/SpendingsSummary';

test('should correctly render SpendingsSummary with 1 spending', () => {
  const wrapper = shallow(<SpendingsSummary spendingsCount={1} spendingsTotal={235}/>)
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render SpendingsSummary with multiple spendings', () => {
  const wrapper = shallow(<SpendingsSummary spendingsCount={1} spendingsTotal={23512340987}/>)
  expect(wrapper).toMatchSnapshot();
});
