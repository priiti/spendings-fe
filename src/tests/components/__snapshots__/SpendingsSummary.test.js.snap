// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SpendingsSummary with 1 spending 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    spending
     totaling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render SpendingsSummary with multiple spendings 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    spending
     totaling 
    $235123409.87
  </h1>
</div>
`;
